docs/apps.md

```diff
@@ -0,0 +1,505 @@
+          "permissions": {
+            "data/io.cozy.emails": {
+              "description": "Requis pour lire et écrire des emails"
+            }
+          }
+        }
+      },
+      "version": "1.2.3",
+      "license": "AGPL-3.0",
+      "permissions": {
+        "data/io.cozy.emails": {
+          "description": "Required for reading and writing emails",
+          "access": "readwrite"
+        }
+      }
+    }
+  }
+}
+```
+
+#### Notes
+
+- The server should keep the manifest in cache, as it will be probably used in
+  the near future to install the application.
+- To start, we will implement a git provider to fetch manifest and install
+  apps. Later, we will add other providers, like mercurial and npm.
+- It's possible to use a branch for git, by putting it the fragment of the
+  URL, like `git://github.com/cozy/cozy-emails#develop`.
+- To download the manifest with git, we can use [git
+  archive](https://www.kernel.org/pub/software/scm/git/docs/git-archive.html),
+  except on github (where it's blocked). For github, we can use
+  `https://raw.githubusercontent.com/:user/:project/:branch/manifest.webapp`
+
+### POST /apps/:slug
+
+Install an application, ie download the files and put them in `/apps/:slug` in
+the virtual file system of the user, create an `io.cozy/apps` document,
+register the permissions, etc.
+
+#### Query-String
+
+Parameter | Description
+----------|-----------------------------------------
+Source    | URL from where the app can be downloaded
+
+#### Request
+
+```http
+POST /apps/emails?Source=git://github.com/cozy/cozy-emails HTTP/1.1
+Accept: application/vnd.api+json
+```
+
+#### Response
+
+```http
+HTTP/1.1 202 Accepted
+Content-Type: application/vnd.api+json
+```
+
+```json
+{
+  "data": [{
+    "id": "4cfbd8be-8968-11e6-9708-ef55b7c20863",
+    "type": "io.cozy.applications",
+    "attributes": {
+      "name": "calendar",
+      "state": "installing",
+      ...
+    }
+  }]
+}
+```
+
+
+List installed applications
+---------------------------
+
+### GET /apps
+
+An application can be in one of these states:
+
+- `ready`, the user can use it
+- `installing`, the installation is running and the app will soon be usable
+- `upgrading`, a new version is being installed
+- `uninstalling`, the app will be removed, and will return to the `available` state.
+
+#### Request
+
+```http
+GET /apps HTTP/1.1
+Accept: application/vnd.api+json
+```
+
+#### Response
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/vnd.api+json
+```
+
+```json
+{
+  "data": [{
+    "id": "4cfbd8be-8968-11e6-9708-ef55b7c20863",
+    "type": "io.cozy.applications",
+    "attributes": {
+      "name": "calendar",
+      "state": "ready",
+      ...
+    }
+  }]
+}
+```
+
+
+Manage the marketplace
+----------------------
+
+### GET /apps/manifests
+
+List applications in the marketplace.
+
+### POST /apps/manifests
+
+Add an application to the marketplace. The payload is a subset of the
```

Comment:
Oh, I thought this will be done internally and didn't see the need for an HTTP route, that does open an interesting topic : will the workers communicate with the stack through http ? Shouldn't we explictly move this kind of route to a separate endpoint `/workerapi` ?